src/main/java/com/google/sps/servlets/SaveQuestionsFromBankServlet.java

```diff
@@ -0,0 +1,126 @@
+  @Override
+  public void doPost(final HttpServletRequest request,
+      final HttpServletResponse response) throws IOException {
+    /*Saves the selected questions from the question bank to the test
+    * that the user wants the questions saved to
+    */
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      logger.atWarning().log("User is not logged in.");
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED,
+          "You are not authorised to view this page");
+      return;
+    }
+    logger.atInfo().log("user=%s is logged in", userService.getCurrentUser());
+    String ownerID = userService.getCurrentUser().getEmail();
+
+    String testName = request.getParameter("test");
+    String[] questionsList = request.getParameterValues("question");
+    if (testName == null || questionsList == null) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      logger.atWarning().log("One or more null parameters");
+      return;
+    }
+    try {
+      for (int i = 0; i < questionsList.length; i++) {
+        addQuestionToExamList(Long.valueOf(questionsList[i]), ownerID, testName);
+        response.getWriter().println("Successfully added Question "
+            + questionsList[i]);
+      }
+      response.sendRedirect("/questionForm");
+    } catch (Exception e) {
+      logger.atSevere().log("There was an error with saving the questions"
+        + " : %s", e);
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+          "Internal Error occurred when trying to save your questions to the test");
+      return;
+    }
+
+  }
+  private void addQuestionToExamList(final long questionEntityKey,
+      final String ownerID, final String testName) {
+    /*Function that adds the question id to the list of questions to the latest
+    *  test created by the user
+    * Arguments
+    * -QuestionEntityKey - id of the question Entity we are adding to the list
+    * -ownerID - email of the person who is adding this question to their test
+    */
+    //grab the latest exam created by the user
+    try {
+      Entity latestExam = getExam(ownerID, testName);
+      if (latestExam.getProperty("questionsList") == null) {
+        List<Long> questionList = new ArrayList<>();
+        questionList.add(questionEntityKey);
+        latestExam.setProperty("questionsList", questionList);
+      } else {
+        List<Long> questionList =
+            (List<Long>) latestExam.getProperty("questionsList");
+        questionList.add(questionEntityKey);
+        latestExam.setProperty("questionsList", questionList);
+      }
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(latestExam);
+      logger.atInfo().log("Added question: %s to test: %s", questionEntityKey,
+          latestExam.getKey().getId());
+    } catch (Exception e) {
```

Comment:
It's probably not a good idea to catch such a general exception. You should only catch exception types that you can handle.

Reply:
https://github.com/googleinterns/step254-2020/issues/38